src/lib/icon/icon-registry.ts

```diff
@@ -116,16 +137,40 @@ export class MatIconRegistry {
     return this.addSvgIconInNamespace('', iconName, url);
   }
 
+  /**
+   * Registers an icon using an HTML string in the default namespace.
+   * @param iconName Name under which the icon should be registered.
+   * @param template SVG source of the icon.
+   */
+  addSvgIconTemplate(iconName: string, template: SafeHtml): this {
```

Comment:
What do you think about `addSvgIconLiteral` or `addSvgIconFromHtmlString`? Template isn't really the right term because the string is totally static / non-templated

Reply:
I like "literal", I wanted to avoid having having HTML in there because SVG is an XML dialect rather an HTML one.

Reply:
"Literal" sounds good to me